compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -220,31 +223,33 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 let fake_place = PlaceRef::new_sized_aligned(cast_ptr, cast, align);
                 Some(bx.load_operand(fake_place).val)
             }
+            OperandValue::ZeroSized => {
+                let OperandValueKind::ZeroSized = operand_kind else {
```

Comment:
It could.  I wrote it this way for structural similarity with the other arms.

Maybe I should have made this be `match (operand.val, operand_kind)` instead?  Unsure.

Reply:
Itś not that big of a deal, both branches are pretty short